omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/PatientIdentifierController1_9Test.java

```diff
@@ -189,4 +191,61 @@ public void shouldReturnTheAuditInfoForTheFullRepresentation() throws Exception
 		
 		assertNotNull(PropertyUtils.getProperty(result, "auditInfo"));
 	}
+
+	@Test
+	public void shouldNotAddIdentifierInUseByAnotherPatient() throws Exception {
+		SimpleObject patientIdentifier = new SimpleObject();
+		patientIdentifier.add("identifier", "123456789qwerty");
+		patientIdentifier.add("identifierType", "2f470aa8-1d73-43b7-81b5-01f0c0dfa53c");
+		patientIdentifier.add("location", RestTestConstants1_8.LOCATION_UUID);
+
+		String json = new ObjectMapper().writeValueAsString(patientIdentifier);
+
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+
+		SimpleObject newPatientIdentifier = deserialize(handle(req));
+
+		assertNotNull(PropertyUtils.getProperty(newPatientIdentifier, "uuid"));
+
+		final String OTHER_PATIENT_UUID = "5946f880-b197-400b-9caa-a3c661d23041";
+		final String REQUEST_URI = "patient/" + OTHER_PATIENT_UUID + "/identifier";
+
+		int otherPatientActiveIdentifiersSize = service.getPatientByUuid(OTHER_PATIENT_UUID).getActiveIdentifiers()
+				.size();
+		req = request(RequestMethod.POST, REQUEST_URI);
+		req.setContent(json.getBytes());
+		try {
+			handle(req);
+			fail();
+		} catch (Exception ex) {
+			assertTrue(ex instanceof APIException);
+		}
+		assertEquals(otherPatientActiveIdentifiersSize, service.getPatientByUuid(OTHER_PATIENT_UUID)
+				.getActiveIdentifiers().size());
+	}
+
+	@Test
+	public void shouldUpdateAnExistingPatientIdentifier() throws Exception {
+		final String patientIdentifierNewValue = "omrs12-34-00";
+		PatientIdentifier patientIdentifier = service.getPatientIdentifierByUuid(getUuid());
+		final String patientIdentifierUuidThatShouldNotChange = patientIdentifier.getUuid();
+
+		assertFalse(patientIdentifierNewValue.equals(patientIdentifier.getIdentifier()));
+
+		SimpleObject simpleObject = new SimpleObject();
+		simpleObject.add("identifier", patientIdentifierNewValue);
+		String json = new ObjectMapper().writeValueAsString(simpleObject);
+
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		req.setContent(json.getBytes());
+
+		SimpleObject updatedPatientIdentifier = deserialize(handle(req));
+		Object uuid = PropertyUtils.getProperty(updatedPatientIdentifier, "uuid");
+		Object identifierValue = PropertyUtils.getProperty(updatedPatientIdentifier, "identifier");
+
+		assertEquals(patientIdentifierUuidThatShouldNotChange, uuid);
+		assertEquals(patientIdentifierNewValue, identifierValue);
+		assertEquals(patientIdentifierNewValue, patientIdentifier.getIdentifier());
+	}
```

Comment:
We _do_ need to add a unit test for this feature, which should be placed in the `EncounterController2_0Test.java" class, which is where the existing tests of the Encounter 2.0 search functionality live.

Reply:
can you give me a test data for the visits as input and the expected output for the encounters for a patient to check for it?